fava/template_filters.py

```diff
@@ -173,20 +173,18 @@ def format_errormsg(message):
     return new_message.replace('for \'', 'for ').replace('\': ', ': ')
 
 
-def group_by_date(entries, kind='M'):
-    if kind == 'Y':
-        def group_fn(d): return d.year
-    elif kind == 'Q':
-        def group_fn(d): return '{}-Q{}'.format(d.year, (d.month-1)//3+1)
-    elif kind == 'M':
-        def group_fn(d): return '{}-{:02}'.format(d.year, d.month)
-    elif kind == 'W':
-        def group_fn(d): return '{}-W{:02}'.format(d.year, d.isocalendar()[1])
-    elif kind == 'D':
-        def group_fn(d): return '{}-{:02}-{:02}'.format(d.year, d.month, d.day)
-    else:
-        return [(None, entries)]
-    values = set(map(lambda entry: group_fn(entry.date), entries))
-    return [
-        (x, [entry for entry in entries if group_fn(entry.date) == x])
-        for x in values]
+def date_group(entry):
+    """Return the date group an entry belongs to."""
+    kind = g.ledger.fava_options['interval']
+    d = entry.date
+    if kind == 'year':
+        return d.year
+    elif kind == 'quarter':
+        return '{}-Q{}'.format(d.year, (d.month-1)//3+1)
```

Comment:
there's no need to render this to a string, just return a tuple

Reply:
or maybe nevermind: for the comparison at least it's not needed, but to render out the headers it is. Might not be worth the time saved to have two functions for this